lib/shopify_cli/theme/extension/syncer/extension_serve_job.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+require "project_types/extension/loaders/project"
+require "project_types/extension/loaders/specification_handler"
+require "shopify_cli/partners_api"
+
+module ShopifyCLI
+  module Theme
+    module Extension
+      class Syncer
+        class ExtensionServeJob < ThreadPool::Job
+          POLL_FREQUENCY = 1 # second
+          PUSH_INTERVAL = 5 # seconds
+
+          RESPONSE_FIELD = %w(data extensionUpdateDraft)
+
+          def initialize(ctx, syncer:, extension:)
+            super(POLL_FREQUENCY)
+
+            @ctx = ctx
+            @extension = extension
```

Comment:
Don't currently use this in the file, but may want this for easy access to extension ID, etc.